### DISCUSSION TITLE
I can't see the preview

### DISCUSSION THREAD

**Discussion by Mapachosa** [thegetty/quire#181] [Get Help] [answered]
Hi everyone! I've just installed Quire on a new Mac, and I can't get the preview to run. When I run "quire preview" or "quire preview --verbose," I see the error message I've pasted below. Has anyone encountered this, and any idea how to fix it? 

ana@MacBook-old ~ % quire preview
node:fs:585
  handleErrorFromBinding(ctx);
  ^Error: ENOENT: no such file or directory, open 'config.yml'
    at Object.openSync (node:fs:585:3)
    at Object.readFileSync (node:fs:453:35)
    at B (/usr/local/lib/node_modules/@thegetty/quire-cli/bin/index.js:2:24620)
    at n.value (/usr/local/lib/node_modules/@thegetty/quire-cli/bin/index.js:2:35619)
    at new n (/usr/local/lib/node_modules/@thegetty/quire-cli/bin/index.js:2:34603)
    at /usr/local/lib/node_modules/@thegetty/quire-cli/bin/index.js:2:50187
    at new Promise (<anonymous>)
    at e.value (/usr/local/lib/node_modules/@thegetty/quire-cli/bin/index.js:2:50153)
    at emit (node:events:394:28)
    at Command.<anonymous> (/usr/local/lib/node_modules/@thegetty/quire-cli/bin/index.js:2:122507)
Emitted 'error' event on n instance at:
    at n.value (/usr/local/lib/node_modules/@thegetty/quire-cli/bin/index.js:2:35652)
    at new n (/usr/local/lib/node_modules/@thegetty/quire-cli/bin/index.js:2:34603)
    [... lines matching original stack trace ...]
    at Command.<anonymous> (/usr/local/lib/node_modules/@thegetty/quire-cli/bin/index.js:2:122507)
    at Command.listener (/usr/local/lib/node_modules/@thegetty/quire-cli/node_modules/commander/index.js:370:29)
    at Command.emit (node:events:394:28)
    at Command.parseArgs (/usr/local/lib/node_modules/@thegetty/quire-cli/node_modules/commander/index.js:892:12) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: 'config.yml'
}

Thanks

**Comment by geealbers** [comment]
Hi! It looks like you’re not running the preview command in your project folder. Type `cd` a space and the name you used for your project. Hit enter. That should move you into your project folder where you can try `quire preview` again. Let us know if that works.

**Reply by Mapachosa** [reply]
Hi Ge!

I'm so sorry for my late awswer. It couldn't run the quire preview command even with the recommendation you gave me.
Therefore I did everything again, but when I install the quire this command line skips me:


quireana@MacBook-old ~ % npm install --global @thegetty/quire-cli
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated axios@0.19.2: Critical security vulnerability fixed in v0.21.1. For more information, see https://github.com/axios/axios/pull/3410

added 613 packages, and audited 614 packages in 1m

33 packages are looking for funding
  run `npm fund` for details

7 vulnerabilities (6 moderate, 1 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.

Then, when I run quire preview inside of my project I get this error:
ana@MacBook-old escritos % quire preview
Launching preview server
ℹ Navigate to http://localhost:1313 to see your changes.
ℹ Press Control+C or type "quire stop" to stop the preview.
node:events:371
      throw er; // Unhandled 'error' event
      ^

Error: spawn node_modules/.bin/webpack ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:282:19)
    at onErrorNT (node:internal/child_process:477:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
Emitted 'error' event on ChildProcess instance at:
    at Process.ChildProcess._handle.onexit (node:internal/child_process:288:12)
    at onErrorNT (node:internal/child_process:477:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn node_modules/.bin/webpack',
  path: 'node_modules/.bin/webpack',
  spawnargs: [
    '--watch',
    '-d',
    '--progress',
    '--color',
    '--config',
    'webpack/webpack.config.dev.js'
  ]
}

I appreciate your comments
thanks

**Reply by Erin-Cecele** [reply]
Hi @Mapachosa. If you run the command `quire install` prior to running `quire preview` that should do the trick. This step is required the first time you work in a new repository. It installs Quire dependencies into that particular project. It only needs to be run once. Let us know if that works!

**Reply by Mapachosa** [reply]
Hi @Erin-Cecele. 
Unfortunately doesn't work. I run `quire install` into of my project but I get several ERROR in the installation.
For example,

npm ERR! code 1
npm ERR! path /Users/ana/escritos/themes/default/node_modules/node-sass
npm ERR! command failed
npm ERR! command sh -c node scripts/build.js
npm ERR! Building: /usr/local/bin/node /Users/ana/escritos/themes/default/node_modules/node-gyp/bin/node-gyp.js rebuild --verbose --libsass_ext= --libsass_cflags= --libsass_ldflags= --libsass_library=
npm ERR!   c++ '-DNODE_GYP_MODULE_NAME=libsass' '-DUSING_UV_SHARED=1' '-DUSING_V8_SHARED=1' '-DV8_DEPRECATION_WARNINGS=1' '-DV8_DEPRECATION_WARNINGS' '-DV8_IMMINENT_DEPRECATION_WARNINGS' '-D_GLIBCXX_USE_CXX11_ABI=1' '-D_DARWIN_USE_64_BIT_INODE=1' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DOPENSSL_NO_PINSHARED' '-DOPENSSL_THREADS' '-DLIBSASS_VERSION="3.5.5"' -I/Users/ana/.node-gyp/16.9.1/include/node -I/Users/ana/.node-gyp/16.9.1/src -I/Users/ana/.node-gyp/16.9.1/deps/openssl/config -I/Users/ana/.node-gyp/16.9.1/deps/openssl/openssl/include -I/Users/ana/.node-gyp/16.9.1/deps/uv/include -I/Users/ana/.node-gyp/16.9.1/deps/zlib -I/Users/ana/.node-gyp/16.9.1/deps/v8/include -I../src/libsass/include  -O3 -gdwarf-2 -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=c++11 -stdlib=libc++ -fno-strict-aliasing -MMD -MF ./Release/.deps/Release/obj.target/libsass/src/libsass/src/ast.o.d.raw   -c -o Release/obj.target/libsass/src/libsass/src/ast.o ../src/libsass/src/ast.cpp

and tha last 

npm ERR! /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/type_traits:776:50: note: 'remove_cv' declared here
npm ERR! template <class _Tp> struct _LIBCPP_TEMPLATE_VIS remove_cv
npm ERR!                                                  ^
npm ERR! 1 error generated.
npm ERR! make: *** [Release/obj.target/binding/src/binding.o] Error 1
npm ERR! gyp ERR! build error 
npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! gyp ERR! stack     at ChildProcess.onExit (/Users/ana/escritos/themes/default/node_modules/node-gyp/lib/build.js:262:23)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:394:28)
npm ERR! gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
npm ERR! gyp ERR! System Darwin 20.6.0
npm ERR! gyp ERR! command "/usr/local/bin/node" "/Users/ana/escritos/themes/default/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
npm ERR! gyp ERR! cwd /Users/ana/escritos/themes/default/node_modules/node-sass
npm ERR! gyp ERR! node -v v16.9.1
npm ERR! gyp ERR! node-gyp -v v3.8.0
npm ERR! gyp ERR! not ok 
npm ERR! Build failed with error code: 1

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/ana/.npm/_logs/2021-09-23T18_29_11_671Z-debug.log
✔ Theme dependencies successfully installed.
✔ Run quire preview to view changes locally.

**Reply by Erin-Cecele** [reply]
Hi @Mapachosa. Run `node --version` and if the version number is 14.17.6 or lower then we suggest starting a new project by running `quire new`. See if you are able to preview that project without any errors.

**Reply by Mapachosa** [reply]
Hi @Erin-Cecele. My versión is 
ana@MacBook-old escritos % node --version
v16.9.1
In this case, what should I do?

**Reply by geealbers** [reply]
Hi @Mapachosa, that's helpful. You'll want to reinstall a lower, more stable version of Node. It's called the LTS version, which stands for "long term support". If you go to [their website](https://nodejs.org/en/) now, you should see a download link for **14.17.6 LTS**. That's the version you want to download and install. It would then be a good idea to run the Quire install again, and then start a fresh Quire project.

**Reply by Mapachosa** [reply]
Hi!!

Thank yoy for your advices! I could run the quire. :)

**Comment by Erin-Cecele** [comment, marked as answer]
Hi @Mapachosa. That's great news. For anyone who encounters a similar problem, as @geealbers points out the error was the result of using the wrong version of Node. If you come across this error, you'll want to reinstall a lower, more stable version of Node, called the LTS version. (LTS stands for "long-term support".) You can go to Node's website https://nodejs.org/en/ and download and install the LTS version as indicated by the arrows in the screenshot below.
![Screen Shot 2021-09-29 at 2 25 06 PM](https://user-images.githubusercontent.com/62075223/135350828-f10a9f69-0e82-47ed-8e7c-d5aef93d2285.png)

**Reply by hughlilly** [reply]
I'd like to add that I've found [the npm package `n`](https://www.npmjs.com/package/n) very useful for managing multiple versions of node, and for switching flawlessly between them. It takes out a lot of the guesswork.
